huxley/core/admin/assignment.py

```diff
@@ -41,69 +41,45 @@ def load(self, request):
-                bad_columns.add(0)
-            return bad_columns
-
-        def create_assignments(reader):
+        def generate_assignments(reader):
             committees = {}
             countries = {}
             schools = {}
-            assigned = set()
-            successful_rows = []
-            failed_rows = []
 
             for row in reader:
                 if (row[0]=='School' and row[1]=='Committee' and row[2]=='Country'):
                     continue # skip the first row if it is a header
-                
+
                 while len(row) < 3:
                     row.append("") # extends the row to have the minimum proper num of columns
-
-                bad_columns = get_bad_columns(row)
-                already_assigned = (row[1], row[2]) in assigned
-                
                 if len(row) < 4:
                     rejected = False # allow for the rejected field to be null
                 else:
                     rejected = (row[3].lower() == 'true') # use the provided value if admin provides it
-                
-                if not bad_columns and not already_assigned:
+
+                try:
                     committee = get_model(Committee, row[1], committees)
                     country = get_model(Country, row[2], countries)
                     school = get_model(School, row[0], schools)
-                    assigned.add((row[1], row[2]))
-                    successful_rows.append((committee.id, country.id, school.id, rejected))
-                else:
-                    for col in bad_columns:
-                        row[col] = row[col] + ' - INVALID'
-                    
-                    if already_assigned:
-                        row[1] = row[1] + ' - ALREADY ASSIGNED'
-                        row[2] = row[2] + ' - ALREADY ASSIGNED'
-                    failed_rows.append(str((row[0], row[1], row[2], str(rejected))))
-            return successful_rows, failed_rows
-
-        successful_rows, failed_rows = create_assignments(reader)
-        if not failed_rows:
-            Assignment.update_assignments(successful_rows)
-            return HttpResponseRedirect(reverse('admin:core_assignment_changelist'))
-        else:
+                    yield (committee.id, country.id, school.id, rejected, True)
+                except ObjectDoesNotExist:
+                    yield (row[1], row[2], row[0], rejected, False)
+
+    
+        failed_rows = Assignment.update_assignments(generate_assignments(reader))
+        if failed_rows:
             messages.error(request, 
-                'Assignments aborted.\nThese are the rows that failed with identification of the cells at fault:\n' + ', \n'.join(failed_rows))
-            return HttpResponseRedirect(reverse('admin:core_assignment_changelist'))
+                'Assignments aborted. These are the rows that failed with identification of the cells at fault: ' + ' | '.join(failed_rows))
```

Comment:
Keep it simple and just say "Assignment upload aborted. These assignments failed: "